File: api-tester/src/main/java/com/revenuecat/apitester/java/PurchasesExceptionAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.PurchasesError;
import com.revenuecat.purchases.PurchasesErrorCode;
import com.revenuecat.purchases.PurchasesException;

final class PurchasesExceptionAPI {
    static void check(final PurchasesException exception) {
        final String underlyingErrorMessage = exception.getUnderlyingErrorMessage();
        final String message = exception.getMessage();
        final PurchasesErrorCode code = exception.getCode();
        final PurchasesError error = exception.getError();
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/PurchasesExceptionAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.PurchasesError
import com.revenuecat.purchases.PurchasesErrorCode
import com.revenuecat.purchases.PurchasesException

class PurchasesExceptionAPI {
    fun check(exception: PurchasesException) {
        val underlyingErrorMessage: String? = exception.underlyingErrorMessage
        val message: String = exception.message
        val code: PurchasesErrorCode = exception.code
        val error: PurchasesError = exception.error
    }
}
